src/status_im2/contexts/communities/overview/view.cljs

```diff
@@ -12,164 +12,95 @@
+(defn open-channel-token-gating-details
+  [name token-gating emoji channel-color]
+  (rf/dispatch
+   [:bottom-sheet/show-sheet
+    {:content
+     (fn []
+       [channel-token-gating-details name token-gating emoji channel-color])
+     :content-height 210}]))
 
-(defn channel-list-component-fn
-  [channel-heights first-channel-height]
-  [rn/view
-   {:on-layout #(swap! first-channel-height
-                  (fn []
-                    (+ (if platform/ios?
-                         0
-                         38)
-                       (int (Math/ceil (oops/oget % "nativeEvent.layout.y"))))))
-    :style     {:margin-top 20 :flex 1}}
-   (map-indexed (fn [index category]
-                  (let [first-category (first category)]
-                    ^{:key first-category}
-                    [rn/view
-                     {:flex      1
-                      :key       (str index first-category)
-                      :on-layout #(swap! channel-heights
-                                    (fn []
-                                      (sort-by :height
-                                               (conj @channel-heights
-                                                     {:height (int (oops/oget % "nativeEvent.layout.y"))
-                                                      :label  first-category}))))}
+(defn layout-y
+  [event]
+  (oops/oget event "nativeEvent.layout.y"))
 
-                     [quo/divider-label
-                      {:label            first-category
-                       :chevron-position :left}]
-                     [rn/view
-                      {:margin-left   8
-                       :margin-top    10
-                       :margin-bottom 8}
-                      (map-indexed (fn [inner-index channel-data]
-                                     [rn/view
-                                      {:key        (str inner-index (:name channel-data))
-                                       :margin-top 4}
-                                      [quo/channel-list-item channel-data]])
-                                   (first-category mock-list-of-channels))]]))
-                mock-list-of-channels)])
+(defn add-category-height
+  [categories-heights category height]
+  (swap! categories-heights
+    (fn []
+      (sort-by :height
+               (conj @categories-heights
+                     {:height height
+                      :label  category})))))
+
+(defn channel-list-component
+  [{:keys [on-categories-heights-changed
+           on-first-channel-height-changed]}
+   channels-list]
+  (let [categories-heights (reagent/atom [])]
+    [rn/view
+     {:on-layout #(on-first-channel-height-changed (+ (if platform/ios?
+                                                        0
+                                                        38)
+                                                      (int (Math/ceil (layout-y %)))))
+      :style     {:margin-top 20 :flex 1}}
+     (map-indexed
```

Comment:
most likely it should be in doall form because there is a deref inside

Reply:
why not a flatlist btw ?

Reply:
Not sure if `doall` is needed. Atom derefs in `:on-layout` callback, so it will be at the moment when item already realized. Or there is something else?

Not sure why not a FlatList.. it is an inherited piece of code, not written by me.

Reply:
but it seems like flatlist should be used here, but not related to this PR